packages/flutter/test/scheduler/scheduler_test.dart

```diff
@@ -237,6 +239,22 @@ void main() {
     warmUpDrawFrame();
     expect(scheduler.hasScheduledFrame, isTrue);
   });
+
+  test('The task was successfully executed after the animation', () async {
+    scheduler.schedulingStrategy = defaultSchedulingStrategy;
+    scheduler.resetEventLoop();
+
+    bool taskExecuted = false;
+    final AnimationController controller = AnimationController(
+      duration: const Duration(milliseconds: 200),
+      vsync: const TestVSync(),
+    );
+    controller.forward();
+
+    // If the task is not scheduled round-robin, this code will block the test, test will be timeout
```

Comment:
Having this time out is not great. Can this run a a fake async zone where we can control time to ensure that the task will execute after the expected time has elapsed?

Reply:
I actually wrote the following code with reference to other tests. Make sure the animation has been performed over time  but it failed .
```dart
 test('The task was successfully executed after the animation', () async {
    scheduler.schedulingStrategy = defaultSchedulingStrategy;
    scheduler.resetEventLoop();

    bool taskExecuted = false;
    final AnimationController controller = AnimationController(
      duration: const Duration(milliseconds: 200),
      vsync: const TestVSync(),
    );
    controller.forward();

    scheduler.scheduleTask(() => taskExecuted = true, Priority.idle);

    tick(const Duration(milliseconds: 100));
    expect(taskExecuted, isFalse);

    tick(const Duration(milliseconds: 300));
    expect(taskExecuted, isTrue);

  });
```
I read the logs and tested the code to execute synchronously, but that wasn't what I wanted.Is there any relevant document or case you can refer to? I will rewrite the test case

Reply:
I have fixed the other issues, but when writing the test using the FakeAsync Zone, the test failed. After debugging, I found that the `Timer. run`  in Fake Async will be immediately added to the `_timers` set, and the tasks in the set will be processed synchronously while `_firetimersWhile`, so the EventLoop will schedule and add the tasks in a cycle. No other code will continue to run. But this is not the same as actually running on the device.

The following related code：
```dart
class FakeAsync {

  /// Creates a new timer controlled by `this` that fires [callback] after
  /// [duration] (or every [duration] if [periodic] is `true`).
  Timer _createTimer(Duration duration, Function callback, bool periodic) {
    var timer = FakeTimer._(duration, callback, periodic, this);
    _timers.add(timer);
    return timer;
  }
}

  /// Invoke the callback for each timer until [predicate] returns `false` for
  /// the next timer that would be fired.
  ///
  /// Microtasks are flushed before and after each timer is fired. Before each
  /// timer fires, [_elapsed] is updated to the appropriate duration.
  void _fireTimersWhile(bool Function(FakeTimer timer) predicate) {
    flushMicrotasks();
    for (;;) {
      if (_timers.isEmpty) break;

      var timer = minBy(_timers, (FakeTimer timer) => timer._nextCall)!;
      if (!predicate(timer)) break;

      _elapseTo(timer._nextCall);
      timer._fire();
      flushMicrotasks();
    }
  }
```

Reply:
cc  when you have time~ thanks  @goderbauer @Hixie

Reply:
I don't fully understand what you mean by this. It looks like the timeout I was concerned about is now gone from the test?

Reply:
If it is not an EventLoop, it appears in the following line of code
  `  await scheduler.scheduleTask(() => taskExecuted = true, Priority.idle);`

Reply:
Because in the original code, once 'schedulingStrategy' has returned the fase, it will not be executed regardless of whether there is a queue left in the `_taskQueue`, so this task will never be executed, causing the test to time out.

Reply:
I tried to use `FakeAsync` to control the animation time, but the `Timer.run` in `FakeAsync` is a synchronous queue, which causes the system to execute `Timer.run(_runTask)` in a loop, thus timed out. So I can only write test cases in this way in the end. If you have better ideas, can you give me some suggestions